packages/vscode-extension/src/debug/teamsfxDebugProvider.ts

```diff
@@ -166,17 +169,22 @@ export class TeamsfxDebugProvider implements vscode.DebugConfigurationProvider {
       }
       debugConfiguration.url = result;
 
-      VsCodeLogInstance.info(sideloadingDisplayMessages.title(debugConfiguration.teamsfxHub!));
-      VsCodeLogInstance.outputChannel.appendLine("");
-      VsCodeLogInstance.outputChannel.appendLine(
-        sideloadingDisplayMessages.sideloadingUrlMessage(
-          debugConfiguration.teamsfxHub!,
-          debugConfiguration.url
-        )
-      );
-      if (isLocal) {
+      // NOTE: handle the case that msedge/chrome will be resolved twice
+      if (debugConfiguration.type.startsWith("pwa")) {
```

Comment:
Would there be risk if vscode updates its resolve logic...
How about checking *url is updated*? For those no-placeholder ones, just ignore since the url is already plain text.

Reply:
Url can contain no placeholders (like Outlook message extension url). Let me add an extra variable to indicate if already resolved.